src/osx/carbon/font.cpp

```diff
@@ -819,6 +819,19 @@ void wxNativeFontInfo::InitFromFontDescriptor(CTFontDescriptorRef desc)
 
     wxCFTypeRef(CTFontDescriptorCopyAttribute(m_descriptor, kCTFontFamilyNameAttribute)).GetValue(m_familyName);
     wxCFTypeRef(CTFontDescriptorCopyAttribute(m_descriptor, kCTFontNameAttribute)).GetValue(m_postScriptName);
+    AdjustPostScriptName();
+}
+
+void wxNativeFontInfo::AdjustPostScriptName()
+{
+    if ( wxCheckOsVersion( 10, 16) )
+    {
+        // the PostScript names reported in macOS start with a dot for System Fonts, this has to be corrected
+        // otherwise round-trips are not possible, resulting in a Times Fallback, therefore we replace these with
+        // their official PostScript Name
+        if ( m_postScriptName.StartsWith(".SFNS") )
+            m_postScriptName.Replace(".SFNS","SFProDisplay");
```

Comment:
In theory, this could replace more than one occurrence of `.SFNS`, should it occur more than once. You could just pass `false` as the last parameter here or do something like
```suggestion
        wxString rest;
        if ( m_postScriptName.StartsWith(".SFNS", &rest) )
            m_postScriptName = "SFProDisplay" + rest;
```